packages/@sanity/base/src/search/weighted/createSearchQuery.ts

```diff
@@ -0,0 +1,99 @@
+import type {ObjectSchemaType} from '@sanity/types'
+import {compact, flatten, flow, toLower, union, uniq} from 'lodash'
+import {joinPath} from '../../util/searchUtils'
+import {tokenize} from '../common/tokenize'
+import {SearchOptions, SearchPath, SearchSpec, SearchTerms, WeightedSearchOptions} from './types'
+
+export interface SearchParams {
+  __types: string[]
+  __limit: number
+  __offset: number
+  [key: string]: unknown
+}
+
+export interface SearchQuery {
+  query: string
+  params: SearchParams
+  options: Record<string, unknown>
+  searchSpec: SearchSpec[]
+  terms: string[]
+}
+
+type ObjectSchema = {
+  name: string
+  // eslint-disable-next-line camelcase
+  __experimental_search: ObjectSchemaType['__experimental_search']
+}
+
+export const DEFAULT_LIMIT = 1000
+
+const combinePaths = flow([flatten, union, compact])
+
+const pathWithMapper = ({mapWith, path}: SearchPath): string =>
+  mapWith ? `${mapWith}(${path})` : path
+
+export function createSearchQuery(
```

Comment:
FYI: Most of this code is hoisted as is from createWeightedSearch.ts to make it unit-testable, and for combining per-search-config with globally provided config. Unlike before, we are now always recreating searchSpec, unlike before when it was done once when creating the weighted search function. This is so we can change which types are included in the query.  

Another new bit is the offset param, which would allow for running paginated searches. It is unused atm.